### PR TITLE
Settings

### PR DESCRIPTION
**Pull request by saarazari5**

### CODE REVIEW

File: app/src/main/java/com/example/whatsapp_clone/Model/Retrofit/HTTPClientDataSource.java [hunk 1]
```diff
@@ -25,7 +25,6 @@ public class HTTPClientDataSource {
     private HTTPClientService service;
```
Comment (oriKatzir) [inline app/src/main/java/com/example/whatsapp_clone/Model/Retrofit/HTTPClientDataSource.java:25]: we need to think of a different name for this, this is not a servicee

File: app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java [hunk 2]
```diff
@@ -141,10 +147,129 @@ public boolean onQueryTextChange(String newText) {
...
 
+
+    private void showSettingsBottomSheet() {
+        View bottomSheetView = getLayoutInflater().inflate(R.layout.settings_bottom_sheet, null);
+        BottomSheetDialog bottomSheetDialog = new BottomSheetDialog(this);
+        bottomSheetDialog.setContentView(bottomSheetView);
+
+        // Calculate the desired height for the bottom sheet
+        int screenHeight = getResources().getDisplayMetrics().heightPixels;
+        int desiredHeight = (int) (screenHeight * 0.75);
```
Comment (oriKatzir) [inline app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java:164]: is this for the up to fit in different screen sizes? will it work for landscape mode? like when the screen is vertical?

File: app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java [hunk 3]
```diff
@@ -141,10 +147,129 @@ public boolean onQueryTextChange(String newText) {
...
+            dialog.dismiss();
+
+            NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+            int firstDestinationId = navController.getGraph().getStartDestination();
+            navController.popBackStack(firstDestinationId, false);
+        });
+
+
+
+    }
```
Comment (oriKatzir) [inline app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java:273]: there is a lot of logic in the main activity, should we move some of it to a utility class or something?

File: app/src/main/java/com/example/whatsapp_clone/Model/Retrofit/HTTPClientDataSource.java (hunk 1 above)
Comment (saarazari5) [reply app/src/main/java/com/example/whatsapp_clone/Model/Retrofit/HTTPClientDataSource.java:25]: it is a service , just not the android service

File: app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java (hunk 2 above)
Comment (saarazari5) [reply app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java:164]: yes the height is determined based on the device orientation

File: app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java (hunk 3 above)
Comment (saarazari5) [reply app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java:273]: this is exactly the job for the main activity. to handle global changes for the entire app, such as pressing a button on the menu and do some UI staff, no extra logic is here

**Pull request merged**